src/renderer/services/wallet/service.test.ts

```diff
@@ -0,0 +1,93 @@
+import { none, some } from 'fp-ts/lib/Option'
+import * as fs from 'fs-extra'
+
+import { removeKeystore, keystoreService, KEY_FILE } from './service'
+
+jest.mock('electron', () => ({ ipcRenderer: { send: jest.fn() } }))
+
+const mockEncrypt = { encrypt: 'info' }
+const mockDecrypt = { phrase: 'phrases' }
+
+jest.mock('@thorchain/asgardex-crypto', () => ({
+  encryptToKeyStore: () => mockEncrypt,
+  decryptFromKeystore: () => Promise.resolve(mockDecrypt.phrase)
+}))
+
+describe('services/keystore/service/', () => {
+  const _setKeystoreStateSpy = jest.spyOn(keystoreService, '_setKeystoreState')
+  const fsRemoveSpy = jest.spyOn(fs, 'remove')
+  const fsWriteJsonSpy = jest.spyOn(fs, 'writeJSON')
+  const fsEnsureFileSpy = jest.spyOn(fs, 'ensureFile')
+  const fsPathExistsSpy = jest.spyOn(fs, 'pathExists')
+  const fsReadJSONSpy = jest.spyOn(fs, 'readJSON')
+
+  beforeEach(() => {
+    fsRemoveSpy.mockImplementationOnce(jest.fn)
+    fsWriteJsonSpy.mockImplementationOnce(jest.fn)
+    fsEnsureFileSpy.mockImplementationOnce(jest.fn)
+  })
+
+  it('removeKeystore', async (done) => {
+    await removeKeystore()
+    expect(fsRemoveSpy).toBeCalledWith(KEY_FILE)
+    expect(_setKeystoreStateSpy).toBeCalledWith(none)
+    done()
+  })
+
+  it('addKeystore', async (done) => {
+    const removeKeystoreSpy = jest.spyOn(keystoreService, 'removeKeystore')
+    const phrase = 'phrase'
+    const password = 'password'
+    removeKeystoreSpy.mockImplementationOnce(() => Promise.resolve())
+
+    await keystoreService.addKeystore(phrase, password)
+
+    expect(removeKeystoreSpy).toBeCalled()
+    expect(fsEnsureFileSpy).toBeCalledWith(KEY_FILE)
+    expect(fsWriteJsonSpy).toBeCalledWith(KEY_FILE, mockEncrypt)
+    expect(_setKeystoreStateSpy).toBeCalledWith(some(some({ phrase })))
+    done()
```

Comment:
Not sure if we really need to check internal things such as "is called with `xyz`". IMO more important for tests are results of a "black box", but not how it comes to its result.

So just a check if `unlock` could be "resolved" in this case ^:
```
// Promise<void> resolves in `undefined`
 await expect(keystoreService.unlock(some(some({ phrase: 'phrase' })), password)).resolves.toBeUndefined()
``` 
And after that we could check other expected states, for example `KeystoreState`.

Reply:
UPDATE: All `toBeCalled` for file actions (remove/fileExists/writeJson)  do make sense for sure! What about the following: Instead of using `setKeystoreStateSpy` just test the `KeystoreState` with expected values, but no check of `_setKeystoreStateSpy.toBeCalledWith`?